### PR TITLE
feat: use the new API for calling functions

### PR DESCRIPTION
**Pull request by its-darsh**

### CODE REVIEW

File: src/vm/builtins/ForeignFunctionInterface.zig [hunk 1]
```diff
@@ -543,93 +543,72 @@ export fn evaluateCallback(cif: [*c]ffi.ffi_cif, maybe_return_address: ?*anyopaq
 
     const vm = maybe_callback_data.?.vm;
     const function = maybe_callback_data.?.function;
+    var args = std.ArrayList(Code.Value).init(vm.allocator);
```
Comment (yhyadev) [inline src/vm/builtins/ForeignFunctionInterface.zig:546]: function_arguments, args is never used across the enrire codebase

**Review by yhyadev — CHANGES_REQUESTED** [review]
Just a little rename

File: src/vm/builtins/ForeignFunctionInterface.zig (hunk 1 above)
Comment (its-darsh) [reply src/vm/builtins/ForeignFunctionInterface.zig]: done

**Review by yhyadev — APPROVED** [review]
Ok

**Pull request merged**